### PR TITLE
fix: solve android foreground application to load received notification

### PR DESCRIPTION
**Pull request by arthursz**
@evandroeisinger 
@tmpapageorgiou 
@lucasts

### CODE REVIEW

**Comment by lucasts** [comment]
@arthursz could you explain your changes and why they work?

**Comment by arthursz** [comment]
the problem is that when the application was running in foreground and a notification was opened a error occurred in react-native because the event was emitted before the application start his lifecycle. I added comments inside the code to explain how we solved the problem.

File: android/src/main/java/com/globo/reactnativeua/ReactNativeUAReceiver.java [hunk 1]
```diff
@@ -1,41 +1,51 @@
...
     protected boolean onNotificationOpened(@NonNull Context context, @NonNull NotificationInfo notificationInfo) {
-        if (ReactNativeUAEventEmitter.getInstance() != null) {
-            ReactNativeUAEventEmitter.getInstance().sendEvent("onNotificationOpened", notificationInfo.getMessage());
-        }
+        Intent intent = new Intent();
+
+        intent.setAction("com.urbanairship.push.RECEIVED");
+        intent.putExtra("com.urbanairship.push.EXTRA_PUSH_MESSAGE_BUNDLE", notificationInfo.getMessage().getPushBundle());
+
+        context.sendBroadcast(intent);
```
Comment (arthursz) [inline android/src/main/java/com/globo/reactnativeua/ReactNativeUAReceiver.java:32]: when a notification is opened we create a intent to simulate Urban Airship push receivement event and send it in broadcast. That way the application will be running when onPushReceived is executed

File: android/src/main/java/com/globo/reactnativeua/ReactNativeUAReceiver.java [hunk 2]
```diff
@@ -1,41 +1,51 @@
...
+        List<ActivityManager.RunningAppProcessInfo> processInfos = activityManager.getRunningAppProcesses();
+        
+        for (ActivityManager.RunningAppProcessInfo processInfo : processInfos) {
+            if (processInfo.processName.equals(context.getPackageName())) {
+                if (processInfo.importance == ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND) {
+                    for (String d: processInfo.pkgList) {
+                        return true;
+                    }
+                }
+            }
```
Comment (arthursz) [inline android/src/main/java/com/globo/reactnativeua/ReactNativeUAReceiver.java:48]: isApplicationRunning verify if application is running the foreground UI. It assumes that the application is running if any process of the application is executing

File: android/src/main/java/com/globo/reactnativeua/ReactNativeUAReceiver.java [hunk 3]
```diff
@@ -1,41 +1,51 @@
...
-        }
-    }
+        boolean isRunning = isApplicationRunning(context);
 
-    @Override
-    protected void onNotificationPosted(@NonNull Context context, @NonNull NotificationInfo notificationInfo) {
-        if (ReactNativeUAEventEmitter.getInstance() != null) {
-            ReactNativeUAEventEmitter.getInstance().sendEvent("onNotificationPosted", notificationInfo.getMessage());
+        if (isRunning && ReactNativeUAEventEmitter.getInstance() != null) {
+            ReactNativeUAEventEmitter.getInstance().sendEvent("onNotificationOpened", message);
```
Comment (arthursz) [inline android/src/main/java/com/globo/reactnativeua/ReactNativeUAReceiver.java:21]: send event to react-native if the emitter is initialized and application is running

**Pull request merged**